flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/classification/MultinomialNaiveBayes.scala

```diff
@@ -0,0 +1,900 @@
+import org.apache.flink.api.scala._
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.core.fs.FileSystem.WriteMode
+import org.apache.flink.ml.common.{ParameterMap, Parameter}
+import org.apache.flink.ml.pipeline.{PredictDataSetOperation, FitOperation, Predictor}
+import org.apache.flink.util.Collector
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.collection.mutable.Map
+
+/**
+ * While building the model different approaches need to be compared.
+ * For that purpose the fitParameters are used. Every possibility that might enhance
+ * the implementation can be chosen separately by using the following list of parameters:
+ *
+ * Possibility 1: way of calculating document count
+ *  P1 = 0 -> use .count() to get count of all documents
+ *  P1 = 1 -> use a reducer and a mapper to create a broadcast data set containing the count of
+ *    all documents
+ *
+ * Possibility 2: all words in class (order of operators)
+ *    If p2 = 1 improves the speed, many other calculations must switch their operators, too.
+ *  P2 = 0 -> first the reducer, than the mapper
+ *  P2 = 1 -> first the mapper, than the reducer
+ *
+ * Possibility 3: way of calculating pwc
+ *  P2 = 0 -> join singleWordsInClass and allWordsInClass to wordsInClass data set
+ *  P3 = 1 -> work on singleWordsInClass data set and broadcast allWordsInClass data set
+ *
+ * Schneider/Rennie 1: ignore/reduce word frequency information
+ *  SR1 = 0 -> word frequency information is not ignored
+ *  SR1 = 1 -> word frequency information is ignored (Schneiders approach)
+ *  SR1 = 2 -> word frequency information is reduced (Rennies approach)
+ *
+ * Schneider1: ignore P(c_j) in cMAP formula
+ *  S1 = 0 -> normal cMAP formula
+ *  S2 = 1 -> cMAP without P(c_j)
+ *
+ * Rennie1: transform document frequency
+ *  R1 = 0 -> normal formula
+ *  R1 = 1 -> apply inverse document frequecy
+ * Note: if R1 = 1 and SR1 = 2, both approaches get applied.
+ *
+ */
+class MultinomialNaiveBayes extends Predictor[MultinomialNaiveBayes] {
+
+  import MultinomialNaiveBayes._
+
+  //The model, that stores all needed information that are related to one specific word
+  var wordRelatedModelData: Option[DataSet[(String, String, Double)]] =
+    None // (class name -> word -> log P(w|c))
+
+  //The model, that stores all needed information that are related to one specifc class+
+  var classRelatedModelData: Option[DataSet[(String, Double, Double)]] =
+    None // (class name -> p(c) -> log p(w|c) not in class)
+
+  //A data set that stores additional needed information for some of the improvements
+  var improvementData: Option[DataSet[(String, Double)]] =
+    None // (word -> log number of documents in all classes / word frequency in all classes
+
+  // ============================== Parameter configuration ========================================
+
+  def setP1(value: Int): MultinomialNaiveBayes = {
```

Comment:
Add documentation for the setter functions, or as in above comment, describe the parameters in class documentation. Have a look at `SVM` class documentation.